packages/components/src/components/CodeEditor/fields.ts

```diff
@@ -0,0 +1,53 @@
+import { StateEffect, StateEffectType, StateField } from "@codemirror/state";
+import { EditorView } from "@codemirror/view";
+import { useEffect } from "react";
+
+import { SqValuePath } from "@quri/squiggle-lang";
+
+import { Simulation } from "../../lib/hooks/useSimulator.js";
+
+// Helper class for turning any React prop into a CodeMirror field.
+// How to use:
+// 1. Create a field once statically with `ReactiveStateField.define<T>(initialValue)`
+// 2. Use `myField.field` as a CodeMirror extension
+// 3. Call `myField.use(view, prop)` as a hook to update the field with an effect.
```

Comment:
I haven't exposed all `<CodeEditor>` props with this class yet, but this approach worked decently for the gutter.

Later, I intend to refactor this class and replace all extension configurations with this pattern (we'll be able to remove `useReactiveExtension`, win some performance, etc.)